### ISSUE TITLE
Support loading environment from .envrc with direnv

### ISSUE THREAD

**Issue by ahgraber** [microsoft/vscode-jupyter#9965]
## Environment data

-   VS Code version: 1.67.0
-   Jupyter Extension version (available under the Extensions sidebar): v2022.4.1011282140
-   Python Extension version (available under the Extensions sidebar): v2022.6.0
-   OS (Windows | Mac | Linux distro) and version: Mac 12.3.1 (MacbookPro 14, M1 Pro)
-   Python and/or Anaconda version: conda 4.11.0, mamba 0.16.0, python 3.9.*/  /3.10.*
-   Type of virtual environment used (N/A | venv | virtualenv | conda | ...): conda
-   Jupyter server running: Local 

## Expected behaviour

If I start an interactive session from a directory with a `.envrc` file that is loaded with [`direnv`](https://direnv.net/), the environmental variables that are loaded with `direnv` should be available via `os.environ`.

## Actual behaviour

None of the variables that are present in my local environment via `direnv` are available.

## Steps to reproduce:

1. assuming `direnv` is installed, create `.envrc` with variables to be loaded and run `direnv allow .`

```sh
# direnv .envrc file
export TEST="this_is_a_test"
```

2. in same directory, create a python file that we will run via the interpreter

```py
#%%
import os

#%%
"TEST" in os.environ
```

3. Run the cells.  The interpreter python environment will not find the env var "TEST", although the variables _will_ be available if we run the same python commands in the terminal python environment

## Additional references
There may be some overlap with the vscode-python extension, where they seem to have deemed this feature "not important" even though it is significantly workflow- and expectation- breaking.

> Thanks for the suggestion! We talked about it with the team and we have unfortunately decided we will not be moving forward with this idea. We think there isn't an enough widespread need for this to warrant the maintenance cost for the feature.
> 
> _Originally posted by @luabud in https://github.com/microsoft/vscode-python/issues/14250#issuecomment-737538950_

Also requested as a feature in https://github.com/microsoft/vscode-python/issues/18220#issue-1088806430, and .envrc/direnv proposed as a solution in https://github.com/microsoft/vscode-python/issues/18250#issuecomment-1038995827

**Comment by rchiodo** [comment]
Thanks for the suggestion. 

My guess is that since nobody else has asked for this in notebooks yet, the odds are we won't be pursuing this. 

However everything here is open source and we gladly except PRs. The tricky part here is that we don't start our kernels using a terminal. So any terminal additions would be difficult to use. 

[.env](https://code.visualstudio.com/docs/python/environments#_environment-variable-definitions-file) file support is supported though. Could you use that instead?

**Comment by ahgraber** [comment]
Yeah, `.env` is certainly an option; however most of our workflows currently use .envrc, so I'll have to figure out a way to keep the two in sync.

Thanks for the quick response.

**Comment by NktBkzn** [comment]
@rchiodo 
I would also love to have this feature. So I am the second one asking:)
I work with a repo which keeps the definition of env variables in .envrc file. I also run some experiments in jupyter lab. 
It would be great to be able to use env variables defined in .envrc file inside notebooks

**Comment by ahgraber** [comment]
@NktBkzn - I've resolved by moving variables needed in the python environment to a `.env` file and then loading that .env file with direnv by adding the following line to the bottom of my `.envrc`.

```
dotenv_if_exists .env
```

It's hacky, but it works... and I think we're a small enough population that microsoft probably won't add this feature.

**Comment by introom** [comment]
> @NktBkzn - I've resolved by moving variables needed in the python environment to a `.env` file and then loading that .env file with direnv by adding the following line to the bottom of my `.envrc`.
> 
> ```
> dotenv_if_exists .env
> ```
> 
> It's hacky, but it works... and I think we're a small enough population that microsoft probably won't add this feature.

@ahgraber Please use this extension: https://github.com/direnv/direnv-vscode

**Comment by emddarn** [comment]
adding `"jupyter.runStartupCommands": [direnv allow .]` in settings seems to be working for me
